src/ssl.c

```diff
@@ -12360,9 +12382,15 @@ int AddSession(WOLFSSL* ssl)
 
 #ifdef SESSION_CERTS
     if (error == 0) {
-        session->chain.count = ssl->session.chain.count;
-        XMEMCPY(session->chain.certs, ssl->session.chain.certs,
-                sizeof(x509_buffer) * MAX_CHAIN_DEPTH);
+    	if (!overwrite || (overwrite && ssl->session.chain.count > 0)) {
```

Comment:
Why overwrite when !overwrite?

Reply:
If we are not overwriting then we are writing to a new session object (or the next one to be used according to `idx = SessionCache[row].nextIdx++;`). Overwriting means that we have found a session with the same `sessionID` and we want to update the data in the old session.